modules/tfe-instance/aws.tf

```diff
@@ -74,82 +74,72 @@ variable "proxy_url" {
   default = ""
 }
 
+variable "internet_access" {
+  default = false
+}
+
 resource "aws_security_group" "ptfe" {
   vpc_id = "${var.vpc_id}"
-  count  = "${var.internal_security_group_id != "" ? 0 : 1}"
-
-  ingress {
-    from_port   = 22
-    to_port     = 22
-    protocol    = "tcp"
-    cidr_blocks = ["0.0.0.0/0"]
-  }
-
-  ingress {
-    from_port   = 8080
-    to_port     = 8080
-    protocol    = "tcp"
-    cidr_blocks = ["0.0.0.0/0"]
-  }
-
-  # TCP All outbound traffic
-  egress {
-    from_port   = 0
-    to_port     = 65535
-    protocol    = "tcp"
-    cidr_blocks = ["0.0.0.0/0"]
-  }
-
-  # UDP All outbound traffic
-  egress {
-    from_port   = 0
-    to_port     = 65535
-    protocol    = "udp"
-    cidr_blocks = ["0.0.0.0/0"]
-  }
-
   tags {
     Name = "terraform-enterprise"
   }
 }
 
-resource "aws_security_group" "ptfe-external" {
-  count  = "${var.external_security_group_id != "" ? 0 : 1}"
-  vpc_id = "${var.vpc_id}"
+// allow elb to talk to the instance
+resource "aws_security_group_rule" "allow_elb_ingress" {
+  type                     = "ingress"
+  from_port                = 8080
+  to_port                  = 8080
+  protocol                 = "tcp"
+  source_security_group_id = "${aws_security_group.ptfe_external.id}"
+  security_group_id        = "${aws_security_group.ptfe.id}"
+}
 
-  ingress {
-    from_port   = 80
-    to_port     = 80
-    protocol    = "tcp"
-    cidr_blocks = ["0.0.0.0/0"]
+resource "aws_security_group" "ptfe_external" {
+  vpc_id = "${var.vpc_id}"
+  tags {
+    Name = "terraform-enterprise-external"
   }
+}
 
-  ingress {
-    from_port   = 443
-    to_port     = 443
-    protocol    = "tcp"
-    cidr_blocks = ["0.0.0.0/0"]
-  }
+resource "aws_security_group_rule" "allow_internet_80" {
+  count = "${var.internet_access ? 1 : 0 }"
+  type            = "ingress"
+  from_port       = 80
+  to_port         = 80
+  protocol        = "tcp"
+  cidr_blocks     = ["0.0.0.0/0"]
+  security_group_id = "${aws_security_group.ptfe_external.id}"
+}
 
-  # TCP All outbound traffic
-  egress {
-    from_port   = 0
-    to_port     = 65535
-    protocol    = "tcp"
-    cidr_blocks = ["0.0.0.0/0"]
-  }
+resource "aws_security_group_rule" "allow_internet_443" {
+  count = "${var.internet_access ? 1 : 0 }"
+  type            = "ingress"
+  from_port       = 443
+  to_port         = 443
+  protocol        = "tcp"
+  cidr_blocks     = ["0.0.0.0/0"]
+  security_group_id = "${aws_security_group.ptfe_external.id}"
+}
 
-  # UDP All outbound traffic
-  egress {
-    from_port   = 0
-    to_port     = 65535
-    protocol    = "udp"
-    cidr_blocks = ["0.0.0.0/0"]
-  }
+resource "aws_security_group_rule" "allow_tcp_egress" {
+  count = 2
```

Comment:
I could create the list and use length here, at the moment this seems pretty static. Preferences?